crates/ruff_python_parser/src/lexer.rs

```diff
@@ -579,6 +718,19 @@ impl<'source> Lexer<'source> {
     // This is the main entry point. Call this function to retrieve the next token.
     // This function is used by the iterator implementation.
     pub fn next_token(&mut self) -> LexResult {
+        if let Some(fstring_context) = self.fstring_stack.last() {
```

Comment:
I dislike adding a stack lookup into the hot path but I don't see a way of how we could avoid it :(

Reply:
Maybe we could utilize `State`? The problem here is that it'll get updated when inside a f-string expression as it's not persistent.